### PR TITLE
Qwen3-Omni][Bugfix] Replace vLLM fused layers with HF-compatible numerics in code predictor

### PR DESCRIPTION
**Pull request by LJH-LBJ**
PLEASE FILL IN THE PR DESCRIPTION HERE ENSURING ALL CHECKLIST ITEMS (AT THE BOTTOM) HAVE BEEN CONSIDERED.

## Purpose
Resolve: https://github.com/vllm-project/vllm-omni/issues/2286/
Fixes audio quality degradation (noise after ~3000 tokens) in Qwen3-Omni's code predictor by replacing vLLM's fused kernels with plain PyTorch equivalents that match HuggingFace reference numerics.

This is the Qwen3-Omni counterpart of PR [#2277 ](https://github.com/vllm-project/vllm-omni/pull/2277) (which fixed the same issue for Qwen3-TTS). The Qwen3-Omni code predictor was written following PR [#1617](https://github.com/vllm-project/vllm-omni/pull/1617)'s approach and inherits the same precision bug.

| Component | Before (precision issue) | After (HF-compatible) |
|---|---|---|
| RMSNorm | vLLM `RMSNorm` (bf16 variance) | Custom `_RMSNorm` (float32 variance) |
| RoPE | vLLM `get_rope` (bf16 cos/sin) | Custom `_RotaryEmbedding` (float32 cos/sin, `torch.autocast(enabled=False)`) |
| QKV projection | `QKVParallelLinear` (fused) | Separate `nn.Linear` for q/k/v |
| MLP | `MergedColumnParallelLinear` + `RowParallelLinear` | Separate `nn.Linear` for gate/up/down |
| torch.compile | `mode="default"` | `options={"epilogue_fusion": False}` |
| position_ids | `[bsz * seq_len]` 1D flat | `[bsz, seq_len]` 2D (HF format) |
| load_weights | stacked_params_mapping remapping | Direct loading (weight names match HF) |

## Test Plan
curl -s http://localhost:46354/v1/chat/completions -H "Content-Type: application/json" -d '{
"model": "/workspace/models/Qwen3-Omni-30B-A3B-Instruct",
"messages": [{"role": "user", "content": "Please write a 5000-word novel."}],
"modalities": ["audio"]
}' | jq -r '.choices[0].message.audio.data' | base64 -d > output2.wav
## Test Result

There is no noise in the audio. It cannot upload because its size is too big

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan. Please provide the test scripts & test commands. Please state the reasons if your codes don't require additional test scripts. For test file guidelines, please check the [test style doc](https://docs.vllm.ai/projects/vllm-omni/en/latest/contributing/ci/tests_style/)
- [ ] The test results. Please paste the results comparison before and after, or the e2e results.
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model. **Please run `mkdocs serve` to sync the documentation editions to `./docs`.**
- [ ] (Optional) Release notes update. If your change is user-facing, please update the release notes draft.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

**Comment by amy-why-3459** [comment]
Please add a nightly-test label.  @Gaohan123 @david6666666 @gcanlin @hsliuustc0106

**Comment by amy-why-3459** [comment]
We are working on an accuracy benchmark. Could you add a long-output use case to monitor this scenario?

**Comment by amy-why-3459** [comment]
@Sy0307 PTAL

**Comment by LJH-LBJ** [comment]
> We are working on an accuracy benchmark. Could you add a long-output use case to monitor this scenario?

Sure, I can do it in next pr.

**Comment by Sy0307** [comment]
LGTM.
